airflow/api_fastapi/execution_api/datamodels/taskinstance.py

```diff
@@ -197,6 +213,7 @@ def ti_state_discriminator(v: dict[str, str] | StrictBaseModel) -> str:
         Annotated[TITargetStatePayload, Tag("_other_")],
         Annotated[TIDeferredStatePayload, Tag("deferred")],
         Annotated[TIRescheduleStatePayload, Tag("up_for_reschedule")],
+        Annotated[TIRetryStatePayload, Tag("up_for_retry")],
```

Comment:
We will also have to add the state to the discriminator here: https://github.com/apache/airflow/pull/47996/files#diff-45c9c5de28078a7e771659267f5d7396e1d37cfd417b0ffc4c637e25c5956569R182-R204, otherwise it will not acknowledge it and fallthrough to "_other_"